impeller/aiks/aiks_unittests.cc

```diff
@@ -4368,5 +4368,28 @@ TEST_P(AiksTest, GaussianBlurAtPeripheryHorizontal) {
   ASSERT_TRUE(OpenPlaygroundHere(canvas.EndRecordingAsPicture()));
 }
 
+TEST_P(AiksTest, GaussianBlurRotatedAndClipped) {
```

Comment:
Can you make this test be more obvious while testing the same thing?  For example, rotate the picture of boston 90 degrees and clip off the left and right sides equally so that we get the square just in the middle?  That way it is easier to validate that it is correct.

Reply:
In order to be a good smoke test for the coverage hint:
* the image should be drawn with a non-aligned basis (rotation or skew),
* the clip should unevenly clip the surface (some sides should be clipped more than the others),
* both the blurred source rectangle and the coverage hint should be non-square rectangles of different size.

I think if the before/after comparison shows obvious differences from the current golden, that'd be enough to be suitably useful and spark investigation?

Reply:
I think that's good when we talk about making sure current behavior is maintained.  The problem is that it's hard to know if this is correct behavior.  I can't read the test and look at the image and pretty confidently know that is the correct rendering.

I think this would be a bit easier to follow if you rotated around the center of the texture.  Rotating around an anchor that isn't the centroid or one of the corners is already hard to calculate in your head.

Reply:
Alrighty, I changed it so the center of the clip is the center of the blurred image and added some notes about the expected output to a comment before the test:

![image](https://github.com/flutter/engine/assets/919017/78efd11c-19eb-4b2a-99fa-d00f383e4ee7)

Reply:
This is what I had in mind.  I think this tests everything you want and is imo way more simple.  I can easily communicate what it should look like:
```c++
TEST_P(AiksTest, GaussianBlurRotatedAndClipped) {
  Canvas canvas;
  std::shared_ptr<Texture> boston = CreateTextureForFixture("boston.jpg");
  Rect bounds =
      Rect::MakeXYWH(0, 0, boston->GetSize().width, boston->GetSize().height);
  Vector2 image_center = Vector2(bounds.GetSize() / 2);
  Paint paint = {.image_filter =
                     ImageFilter::MakeBlur(Sigma(20.0), Sigma(20.0),
                                           FilterContents::BlurStyle::kNormal,
                                           Entity::TileMode::kDecal)};
  Vector2 clip_size = {200, 100};
  canvas.Scale(GetContentScale());
  canvas.ClipRect(Rect::MakeLTRB(512, 384, 512, 384).Expand(clip_size));
  canvas.Translate({512, 384, 0});
  canvas.Scale({0.5, 0.5, 1});
  canvas.Rotate(Degrees(25));

  canvas.DrawImageRect(std::make_shared<Image>(boston), /*source=*/bounds,
                       /*dest=*/bounds.Shift(-image_center), paint);

  ASSERT_TRUE(OpenPlaygroundHere(canvas.EndRecordingAsPicture()));
}
```

Reply:
sgtm, I went with this and adjusted the clip slightly to make it cover a smaller area of the image.